bsps/shared/start/stackalloc.c

```diff
@@ -43,15 +44,19 @@ void bsp_stack_allocate_init(size_t stack_space_size)
 void *bsp_stack_allocate(size_t size)
 {
   void *stack = NULL;
-
+  uint32_t *page_table_base;
   if (bsp_stack_heap.area_begin != 0) {
     stack = _Heap_Allocate(&bsp_stack_heap, size);
   }
 
   if (stack == NULL) {
     stack = _Workspace_Allocate(size);
   }
-
+  
+  page_table_base = 0x1000;   // This is hard-coded page table base address
+  prot_stack_allocate(stack, size, page_table_base);  // The current way to allocate protected stack is to assign memory attributes 
```

Comment:
You may want to separate the calls to the allocated stack memory and setting the permissions on it, that's you call `prot_stack_allocate` from where `bsp_stack_allocate_init` is called. Otherwise, it will be confusing how to get the `page_table_base` from

Reply:
I am a bit confused about this, `prot_stack_allocate` registers the attributes of the allocated stack to a chain, if we call it from `bsp_stack_allocate_init` how do we register the stack attributes structure to the chain? I agree we should definitely not pass the `page_table_base` from here, at least not in this format. I suppose I will have to think about as to how to decide what the `page_table_base` value for each stack should be.